### PR TITLE
[prompty] Fix prompt config override does not take effect during batch run

### PR DESCRIPTION
**Pull request by lalala123123**
# Description
```python
prompty = load_flow(
    source=f"{PROMPTY_DIR}/prompty_example_without_config.prompty", model={"configuration": config}
)
run = pf.run(flow=prompty, data=f"{DATA_DIR}/prompty_inputs.jsonl")
```
Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/promptflow/_sdk/_pf_client.py [hunk 1]
```diff
@@ -181,9 +181,10 @@ def _run(
             raise ValueError("at least one of data or run must be provided")
 
         is_flow_object = isinstance(flow, FlowBase)
-        if not is_flow_object and callable(flow) and not inspect.isclass(flow) and not inspect.isfunction(flow):
+        if callable(flow) and not inspect.isclass(flow) and not inspect.isfunction(flow):
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_pf_client.py:184]: add some comment here on this check

Comment (lalala123123) [reply src/promptflow-devkit/promptflow/_sdk/_pf_client.py:184]: add the comment

File: src/promptflow-core/promptflow/executor/_prompty_executor.py [hunk 2]
```diff
@@ -20,7 +20,7 @@ class PromptyExecutor(ScriptExecutor):
 
     def __init__(
         self,
-        flow_file: Path,
+        flow_file: Union[Path, Callable],
```
Comment (wangchao1230) [inline src/promptflow-core/promptflow/executor/_prompty_executor.py:23]: maybe rename this param, flow_file is confusing to accept Callable

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow-core/promptflow/executor/_prompty_executor.py (hunk 2 above)
Comment (lalala123123) [reply src/promptflow-core/promptflow/executor/_prompty_executor.py:23]: This is inherited from the naming in ScriptExecutor

Comment (wangchao1230) [reply src/promptflow-core/promptflow/executor/_prompty_executor.py:23]: maybe change in future

File: src/promptflow-core/promptflow/executor/flow_executor.py [hunk 3]
```diff
@@ -214,9 +215,13 @@ def create(
             setup_exporter_from_environ()
 
         if hasattr(flow_file, "__call__") or inspect.isfunction(flow_file):
+            from ._prompty_executor import PromptyExecutor
             from ._script_executor import ScriptExecutor
 
-            return ScriptExecutor(flow_file, connections=connections, storage=storage)
+            if isinstance(flow_file, Prompty):
```
Comment (thy09) [inline src/promptflow-core/promptflow/executor/flow_executor.py:221]: Maybe we can do like this?
if isinstance(flow_file, Prompty)
    return
if hasattr()
    return

Comment (lalala123123) [reply src/promptflow-core/promptflow/executor/flow_executor.py]: fix it

**Review by thy09 — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**